link.go

```diff
@@ -122,13 +119,11 @@ func (l *link) attach(ctx context.Context, beforeAttach func(*frames.PerformAtta
 		// wait for detach
 		select {
 		case <-ctx.Done():
-			// if we don't send an ack then we're in violation of the protocol
-			go func() {
-				ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-				defer cancel()
-				l.muxClose(ctx, nil, nil, nil)
-			}()
-			return ctx.Err()
+			// we are now in an inconsistent state, so kill the connection
+			if err := l.session.conn.Close(); err != nil {
```

Comment:
For the future, I wonder if we'll want to make this kind of behavior optional, similar to the option we have where you can detach on disposition errors.